docs/renovate/readme.md

```diff
@@ -60,3 +60,102 @@ module.exports = {
   ]
 };
 ```
+
+### Complex case with private registries
+
+If some of your dependencies are coming from a private registry, you will have to add the token in the `.npmrc` or `.yarnrc.yml` to be able to fetch the new versions and trigger the install in the post-upgrade commands from the presets.
+However, it is not recommended to put the tokens directly in the code of your repository, instead you should use secrets in your CI.
+
+A solution can be to set up the authentication at user level in the CI (`~/.npmrc`).
+This is only working when running `npx renovate` directly (instead of using the GitHub action which relies on docker images).
+
+Example of setup with npm:
+
+Assuming you created the following secrets in your repository:
+- `PRIVATE_REGISTRY_READ_TOKEN` that will contain a valid read-only token to access your private registry.
+- `RENOVATE_GITHUB_TOKEN` that will contain a valid write token to access your repository to create the pull-requests.
+
+```yaml
+# .github/workflows/renovate.yaml
+name: Self-Hosted Renovate
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 2 * * *'
+env:
+  PRIVATE_REGISTRY_READ_TOKEN: ${{ secrets.PRIVATE_REGISTRY_READ_TOKEN }}
+
+jobs:
+  renovate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - uses: actions/setup-node@0a44ba7841725637a19e28fa30b79a866c81b0a6 # v4.0.4
+        with:
+          node-version: 20
+      - name: Prepare global npmrc for renovate
+        shell: 'bash'
+        run: 'cat .npmrc.ci > ~/.npmrc'
+      - name: 'Running Renovate'
+        run: |
+          git config --global user.email 'bot@renovateapp.com'
+          git config --global user.name 'Renovate Bot'
+          git config --global core.longpaths true
+          cd .github/workflows && npx renovate@~37.420.1
+        env:
+          RENOVATE_TOKEN: ${{ secrets.RENOVATE_GITHUB_TOKEN }}
+          GITHUB_COM_TOKEN: ${{ secrets.RENOVATE_GITHUB_TOKEN }}
+          PRIVATE_REGISTRY_READ_TOKEN: ${{ secrets.PRIVATE_REGISTRY_READ_TOKEN }}
+```
+
+```js
+// .github/workflows/config.js
+module.exports = {
+  gitAuthor: 'Renovate Bot <bot@renovateapp.com>',
+  onboarding: false,
+  platform: 'github',
+  repositories: [
+    // List of repositories to scan
+    // ...
+  ],
+  customEnvVariables: {
+    CI: 'true',
+    PRIVATE_REGISTRY_READ_TOKEN: process.env.PRIVATE_REGISTRY_READ_TOKEN
+  },
+  hostRules: [
```

Comment:
It could be interesting to mention that the hostRules is to allow the Renovate action while the env is for the postUpgradeTask (if I remember well :D)

Reply:
I am not even sure that the `hostRules` are still needed now that we set the authentication globally in `~/.npmrc`

Reply:
after some tests, it turns out that this is very much needed :|
I've updated the doc accordingly